ports/nrf/modules/machine/pin.c

```diff
@@ -105,6 +113,12 @@ STATIC bool pin_class_debug;
 void pin_init0(void) {
     MP_STATE_PORT(pin_class_mapper) = mp_const_none;
     MP_STATE_PORT(pin_class_map_dict) = mp_const_none;
+    MP_STATE_PORT(machine_pin_irq_handlers) = &irq_handlers[0];
+
+    // Initialize GPIOTE if not done yet.
+    if (nrfx_gpiote_is_init() == false) {
```

Comment:
1. Why not just `if (!nrfx_gpiote_is_init())`
2. I think you can skip this check entirely - it is already checked at initialization (but logs an error if enable)

Reply:
1. I got a compiler warning telling me that the return value was not used. But it seems to have gone away now. I'll efix this.
2. If CTRL+D is hit (soft-reset) this module is not re-initilalized giving errors. The same applies to registered pins.